src/pimp_mixer_arm.S

```diff
@@ -214,13 +213,14 @@ pimp_mixer_mix_samples:
 	ldr r4, =0x4000208 @ load address of REG_IME
 	ldr r5, .ime_store @ stash for later
 	str r5, [r4]       @ write value to REG_IME
+
+	ldr sp, .stack_store    @ restore stack pointer
 #endif
 	
 .ret:
 	@ clean return
 	mov r0, SAMPLE_CURSOR
 	
-	ldr sp, .stack_store    @ restore stack pointer
```

Comment:
This one is technically correct, and I think we should just do it. But there's one thing that bothers me: It's asymmetrical. Before, we had one store at entry and one restore at exit. Now we omit the restore when we terminate after the "fixup"-process (by branching to .ret). But it's not *incorrect*, because we haven't clobbered the stack-pointer yet.

Reply:
By "I think we should just do it" do you mean removing this line, or restoring sp (i.e. keeping the line)? Or should I insert an `ldreq ...` before line 135? Or move the `sp`-saving code to line 136 or sth?

Reply:
I mean just apply this part of the code as-is. Moving the stack-pointer saving until after that return is probably the better solution, but I get rather dizzy reading this code ;)